### ISSUE TITLE
[HOLD for payment 2024-03-13] [Fire Cleanup] Upgrade react-navigation to fix potential crashes

### ISSUE THREAD

**Issue by roryabraham** [Expensify/App#34523]
TL;DR #fireroom-2024-01-11-ipad-login-crash was caused by a [bug in react-navigation](https://github.com/react-navigation/react-navigation/issues/11701).

We have asked Callstack to fix this bug, and they are working on it. This is just a tracking issue to make sure we upgrade E/App's react-navigation version after that bug is fixed.

So for now it's on HOLD for https://github.com/react-navigation/react-navigation/issues/11701

<details><summary>Issue Owner</summary>Current Issue Owner: @miljakljajic</details>

**Comment by MrRefactor** [comment]
Hi! Its Kuba from Callstack, I will be working on fixing this bug in react navigation.

**Comment by dangrous** [comment]
Great! Let me know how I can help.

**Comment by MrRefactor** [comment]
Sure thing!

**Comment by dangrous** [comment]
hey @MrRefactor figured I'd check in - how are things going?

**Comment by MrRefactor** [comment]
Hey @dangrous Im still working on the fix in react-navigation repo

**Comment by dangrous** [comment]
@MrRefactor Do you have a rough ETA for the solution? I know these things can be finicky so doesn't have to be exact. Thanks!

**Comment by MrRefactor** [comment]
Hey @dangrous Im planning to create a PR end of this week or early next week!

**Comment by dangrous** [comment]
Sounds great! Looking forward to it.

**Comment by dangrous** [comment]
Hey @MrRefactor checking in - let me know if I can help/review/anything!

**Comment by MrRefactor** [comment]
Hey @dangrous sorry for the delay but I needed to resolve another issue which was on my plate. Ive been talking to Satya about this one and I will create PR this week which will be approved by him next week.

**Comment by MrRefactor** [comment]
Hey @dangrous [PR](https://github.com/react-navigation/react-navigation/pull/11827) is ready and will be check by Satya

**Comment by dangrous** [comment]
Nice! Looking forward to seeing it

**Comment by dangrous** [comment]
Anything I can help with to keep this moving along? Thanks!

**Comment by MrRefactor** [comment]
Waiting on PR being approved and merged by Satya, most likely tomorrow

**Comment by MrRefactor** [comment]
Released new version as in [this](https://github.com/react-navigation/react-navigation/commit/13d4aa270b301faf07960b4cd861ffc91e9b2c46) commit, I will create a PR with bumping up the version in the repo!

@dangrous

**Comment by MrRefactor** [comment]
@dangrous I have created draft PR but right now we are facing typecheck errors, Im trying to resolve those but could you also take a look on those?

**Comment by roryabraham** [comment]
@dangrous has gone on parental leave, I'll step in to help out here

**Comment by roryabraham** [comment]
@MrRefactor It looks like the typecheck errors we're seeing are coming from your PR: https://github.com/react-navigation/react-navigation/pull/11827/files

**Comment by roryabraham** [comment]
You changed the `navigation.getState` prop to return `NavigationState | undefined`, where before it was `NavigationState`. Updating that will require some careful and thoughtful refactoring of our codebase.

I started looking into this but it's pretty late for me. @MrRefactor if you're stuck, maybe ask for help in #expensify-open-source?

**Comment by MrRefactor** [comment]
> You changed the `navigation.getState` prop to return `NavigationState | undefined`, where before it was `NavigationState`. Updating that will require some careful and thoughtful refactoring of our codebase.
> 

Yes - exactly thats the case because getState can return undefined when called outside of the screen. In that case I will post on #expensify-open-source

**Comment by roryabraham** [comment]
PR merged. Adding Bug so we can make payment to @paultsimura for the C+ review

**Comment by roryabraham** [comment]
@miljakljajic all that will be needed here is a C+ payment after https://github.com/Expensify/App/pull/37194 has been on prod for a week

**Comment by miljakljajic** [comment]
Thank you!

**Comment by roryabraham** [comment]
@miljakljajic looks like this shuold be ready to pay out.

@paultsimura please complete the BZ checklist above

**Comment by miljakljajic** [comment]
Paying now!

**Comment by miljakljajic** [comment]
Thank you! @paultsimura - contract sent :)
